src/app/course/course-item/course-item.component.ts

```diff
@@ -9,8 +10,8 @@ import { CourseInstance } from '../course';
 })
 
 export class CourseItemComponent  {
-  @Input() public course: CourseInstance;
-  @Output() public deleteCourse = new EventEmitter();
+  @Input() public course: Course;
+  @Output() public deleteCourse: EventEmitter<number> = new EventEmitter<number>();

```

Comment:
You can omit the property's type definition. Typescript is smart enough to infer right type. The short form would be
```typescript
@Output() public deleteCourse = new EventEmitter<number>();
```